common/beerocks/hostapd/configuration.cpp

```diff
@@ -150,6 +150,23 @@ bool Configuration::set_create_head_value(const std::string &key, const int valu
     return set_create_head_value(key, std::to_string(value));
 }
 
+std::string Configuration::get_head_value(const std::string &key)
+{
+    std::string key_eq(key + "=");
+    auto line_iter = std::find_if(
+        m_hostapd_config_head.begin(), m_hostapd_config_head.end(),
+        [&key_eq, this](const std::string &line) -> bool { return is_key_in_line(line, key_eq); });
+
+    if (line_iter == m_hostapd_config_head.end()) {
+        m_last_message = std::string(__FUNCTION__) +
+                         " couldn't find requested key in head: " + key;
+        return "";
```

Comment:
Shouldn't you set `m_ok = false` before returning?
Also, if function succeeds, `m_last_message` should be cleared.
If `m_last_message` and `m_ok` are updated in parallel, you might infer `m_ok` from `m_last_message`